pkg/kvstore/allocator/allocator_test.go

```diff
@@ -16,526 +16,411 @@
-
-	tests := []struct {
-		prefix   string
-		key      string
-		expected bool
-	}{
-		{
-			prefix:   "foo",
-			key:      "foo/bar",
-			expected: true,
-		},
-		{
-			prefix:   "foo/;bar;baz;/;a;",
-			key:      "foo/;bar;baz;/;a;/alice",
-			expected: true,
-		},
-		{
-			prefix:   "foo/;bar;baz;",
-			key:      "foo/;bar;baz;/;a;/alice",
-			expected: false,
-		},
-		{
-			prefix:   "foo/;bar;baz;/;a;/baz",
-			key:      "foo/;bar;baz;/;a;/alice",
-			expected: false,
-		},
-	}
-
-	for _, tt := range tests {
-		c.Logf("prefixMatchesKey(%q, %q) expected to be %t", tt.prefix, tt.key, tt.expected)
-		result := prefixMatchesKey(tt.prefix, tt.key)
-		c.Assert(result, Equals, tt.expected)
-	}
-}
-
-func (s *AllocatorSuite) TestGetNoCache(c *C) {
-	testGetNoCache(c, idpool.ID(256), randomTestName(), "a") // enable use of local cache
-}
-
-func (s *AllocatorSuite) TestRemoteCache(c *C) {
-	testName := randomTestName()
-	allocator, err := NewAllocator(testName, TestType(""), WithMax(idpool.ID(256)), WithSuffix("a"))
-	c.Assert(err, IsNil)
-	c.Assert(allocator, Not(IsNil))
-
-	// remove any keys which might be leftover
-	allocator.DeleteAllKeys()
-
-	// allocate all available IDs
-	for i := idpool.ID(1); i <= idpool.ID(4); i++ {
-		key := TestType(fmt.Sprintf("key%04d", i))
-		_, _, err := allocator.Allocate(context.Background(), key)
-		c.Assert(err, IsNil)
-	}
-
-	// wait for main cache to be populated
-	c.Assert(testutils.WaitUntil(func() bool { return len(allocator.mainCache.cache) == 4 }, 5*time.Second), IsNil)
-
-	// count identical allocations returned
-	cache := map[idpool.ID]int{}
-	allocator.ForeachCache(func(id idpool.ID, val AllocatorKey) {
-		cache[id]++
-	})
-
-	// ForeachCache must have returned 4 allocations all unique
-	c.Assert(len(cache), Equals, 4)
-	for i := range cache {
-		c.Assert(cache[i], Equals, 1)
-	}
-
-	// watch the prefix in the same kvstore via a 2nd watcher
-	rc := allocator.WatchRemoteKVStore(kvstore.Client(), testName)
-	c.Assert(rc, Not(IsNil))
-
-	// wait for remote cache to be populated
-	c.Assert(testutils.WaitUntil(func() bool { return len(rc.cache.cache) == 4 }, 5*time.Second), IsNil)
-
-	// count the allocations in the main cache *AND* the remote cache
-	cache = map[idpool.ID]int{}
-	allocator.ForeachCache(func(id idpool.ID, val AllocatorKey) {
-		cache[id]++
-	})
-
-	// Foreach must have returned 4 allocations each duplicated, once in
-	// the main cache, once in the remote cache
-	c.Assert(len(cache), Equals, 4)
-	for i := range cache {
-		c.Assert(cache[i], Equals, 2)
-	}
-
-	rc.Close()
-
-	allocator.DeleteAllKeys()
-	allocator.Delete()
-}
-
-// The following tests are currently disabled as they are not 100% reliable in
-// the Jenkins CI
 //
-//func testParallelAllocator(c *C, maxID idpool.ID, allocatorName string, suffix string) {
-//	allocator, err := NewAllocator(allocatorName, TestType(""), WithMax(maxID), WithSuffix(suffix))
+//type AllocatorEtcdSuite struct {
```

Comment:
Add a comment indicating why these are commented out. It should be a priority for us to re-enable these tests somehow soon so we don't lose critical coverage for the kvstore.

Reply:
I copied the comment that was already there. I actually don't know what was exactly wrong, but they are ""not 100% reliable". I'll include a comment that they were copied over and may no longer even compile. I looked through them and they would need a rewrite of the dummyBackend because they test multi-node access. I'll try to sort this out on the plane.